includes/fraud-prevention/class-fraud-prevention-service.php

```diff
@@ -103,6 +103,18 @@ public static function maybe_append_fraud_prevention_token() {
 		);
 	}
 
+	/**
+	 * Checks if current page is the Pay for order page.
+	 *
+	 * @return bool
+	 */
+	public function is_pay_for_order_page() {
+		global $wp;
+		$order_id = $wp->query_vars['order-pay'] ?? null;
+		// phpcs:disable WordPress.Security.NonceVerification
+		return isset( $_GET['pay_for_order'] ) && isset( $_GET['key'] ) && current_user_can( 'pay_for_order', $order_id );
+	}
```

Comment:
There's currently a simpler implementation of this for ECE ([src](https://github.com/Automattic/woocommerce-payments/blob/76d2cd5199a6cfa61b4a36f42ade13af0bb2a78a/includes/express-checkout/class-wc-payments-express-checkout-button-helper.php#L207-L209)). Should we keep both consistent?

Reply:
done https://github.com/Automattic/woocommerce-payments/pull/9089/commits/452881d4cd5e49aeca682091bd0da8536c6c9e8e